### PR TITLE
Add analytical events

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
Not sure a good way to QA this. You will need to deploy a this branch with `jujugui.GTM_enabled = True` then track the calls in the inspector.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6343/
Test FAILed.

File: jujugui/static/gui/src/app/components/notification-list/notification-list.js [hunk 1]
```diff
@@ -56,6 +56,19 @@ YUI.add('notification-list', function() {
         type: notification.level,
         timestamp: notification.timestamp
       };
+
+      // Emiting a google tag manager event registering the notification.
+      /* global dataLayer */
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/notification-list/notification-list.js]: the globals are defined in the `eslint.rc` file. With that said however `dataLayer` is quite generic, is there any way we can make this more unique? I'm guessing that this is defined somewhere out of our control as it's not defined in this PR?

File: jujugui/static/gui/src/app/state/state.js [hunk 2]
```diff
@@ -416,6 +416,19 @@ const State = class State {
 
     this._appStateHistory.push(purgedState);
     this._pushState();
+
+    // Emiting a google tag manager event registering the state change.
+    /* global dataLayer */
```
Comment (hatched) [inline jujugui/static/gui/src/app/state/state.js]: See above about globals.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6345/
Test PASSed.

**Comment by anthonydillon** [comment]
Sorry @hatched forgot you did look at this and I done some more on it. Ready for another look.

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6553/

**Comment by jrwren** [comment]
test this please

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6554/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6352/
Test PASSed.

**Comment by jrwren** [comment]
jenkins: test this please

**Comment by jrwren** [comment]
jenkins: test this please

**Comment by jrwren** [comment]
jenkins: retest this please

**Comment by jrwren** [comment]
jenkins: retest this

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6555/

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**